llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -996,6 +989,30 @@ static Value *canonicalizeSaturatedAdd(ICmpInst *Cmp, Value *TVal, Value *FVal,
   if (!match(TVal, m_AllOnes()))
     return nullptr;
 
+  if ((Pred == ICmpInst::ICMP_UGE || Pred == ICmpInst::ICMP_UGT) &&
+      match(FVal, m_Add(m_Specific(Cmp0), m_APIntAllowPoison(C))) &&
+      match(Cmp1, m_SpecificIntAllowPoison(~*C))) {
+    // (X u> ~C) ? -1 : (X + C) --> uadd.sat(X, C)
```

Comment:
need a comment for `(X u>= ~C) ? -1 : (X + C) --> uadd.sat(X, C)` too

Reply:
> need a comment for `(X u>= ~C) ? -1 : (X + C) --> uadd.sat(X, C)` too

Done!